pkg/logs/input/docker/tailer.go

```diff
@@ -185,19 +205,25 @@ func (t *Tailer) readForever() {
 					// This error is raised when the agent is stopping
 					return
 				case err == io.EOF:
-					// This error is raised when the container is stopping
-					// or when the container has not started to output logs yet.
-					// Retry to read to make sure all logs are collected
-					// or stop reading on the next iteration
-					// if the tailer has been stopped.
-					log.Debugf("No new logs are available for container %v", ShortContainerID(t.ContainerID))
+					// This error is raised when:
+					// * the container is stopping.
+					// * when the container has not started to output logs yet.
+					// * during a file rotation.
+					// restart the reader (by providing the error the t.reader)
+					// the reader will be restarted with a backoff policy.
+					t.source.Status.Error(fmt.Errorf("log decoder returns an EOF error that will trigger a Reader restart"))
+					log.Debugf("log decoder returns an EOF error that will trigger a Reader restart for container %v", ShortContainerID(t.ContainerID))
+					t.reader.err = err
+					continue
 				default:
 					t.source.Status.Error(err)
 					log.Errorf("Could not tail logs for container %v: %v", ShortContainerID(t.ContainerID), err)
 					t.erroredContainerID <- t.ContainerID
 					return
 				}
 			}
+			t.reader.Success()
+			t.source.Status.Success()
```

Comment:
this line could be moved to https://github.com/DataDog/datadog-agent/pull/4860/files?file-filters%5B%5D=.go#diff-d294b0babcfd0e88873fa5983963fb9cR195